# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/icu4c/hotfix-endianess.diff
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- source/tools/pkgdata/pkgdata.cpp	2023-12-09 03:30:05.000000000 +0100
+++ source/tools/pkgdata/pkgdata.cpp	2024-02-21 21:50:02.713084994 +0100
@@ -615,7 +615,7 @@
         if(o->verbose) {
           fprintf(stdout, "# Writing package file %s ..\n", datFileNamePath);
         }
-        result = writePackageDatFile(datFileNamePath, o->comment, o->srcDir, o->fileListFiles->str, nullptr, U_CHARSET_FAMILY ? 'e' :  U_IS_BIG_ENDIAN ? 'b' : 'l');
+        result = writePackageDatFile(datFileNamePath, o->comment, o->srcDir, o->fileListFiles->str, nullptr, U_CHARSET_FAMILY ? 'e' :  U_IS_BIG_ENDIAN ? 'l' : 'b');
         if (result != 0) {
             fprintf(stderr,"Error writing package dat file.\n");
             return result;
--- source/tools/toolutil/pkg_genc.cpp	2023-12-09 03:30:05.000000000 +0100
+++ source/tools/toolutil/pkg_genc.cpp	2024-02-21 21:40:48.414138403 +0100
@@ -620,6 +621,8 @@
         column=1;
     }
 
+    bitField = __builtin_bswap32(bitField);
+
     if (bitField < 10) {
         /* It's a small number. Don't waste the space for 0x */
         *(s++)=hexToStr[bitField];
Only in icu-release-74-2.fixed/icu4c/source/tools/tzcode: Makefile
